### ISSUE TITLE
Add documentation for how to use iis_site options correctly (PreloadEnabled specifically)

### ISSUE THREAD

**Issue by mkeller25** [chef-cookbooks/iis#381]
### Cookbook version
6.7.1

### Chef-client version
12.19.36

### Platform Details
Windows Server 2016

### Scenario:
Trying to properly set PreloadEnabled for our site

### Steps to Reproduce:
iis_site 'mysite.com' do
	protocol :http
	port 80
	path "#{node['iis']['docroot']}\\dataverify"
	application_pool 'dataverify.com'
	**options "-preloadenabled"**
	action [:add, :start, :config]
end

### Expected Result:
I'm expecting the site to have the setting Preload Enabled = True however I cannot seem to get it to work. I'm assuming this would be under "options" though I suppose it may not be supported. It still isn't clear how other options should be formatted.

Preload Enabled is an advanced site setting that is part of the "Application Initialization" windows feature included as part of the larger Windows IIS/Webserver role, as long as you make the selection when you install it. The Chef code looks like

windows_feature ['Web-App-Dev', 'Web-Asp-Net', 'Web-Net-Ext', 'Web-ASP', 'Web-CGI', 'Web-ISAPI-Ext', 'Web-ISAPI-Filter', 'Web-AppInit'] do
	action :install
	install_method :windows_feature_powershell
end

Web-AppInit is the module/feature in question 

### Actual Result:
Nothing. If I pass in

options "preloadenabled" 
or "-preloadenabled" 
or seemingly anything in a string, my recipe still runs just fine but no change gets pushed to the server that I can see.

If I pass in

options -preloadenabled
or options preloadenabled

the recipe will fail to run

**Comment by EasyAsABC123** [comment]
there is no `options` property on the `iis_site` resource. 

here are the properties:
```ruby
property    :site_name, String, name_property: true
property    :site_id, Integer
property    :port, Integer, default: 80, coerce: proc { |v| v.to_i }
property    :path, String
property    :protocol, [Symbol, String], equal_to: [:http, :https, :ftp], default: :http, coerce: proc { |v| v.to_sym }
property    :host_header, String
property    :bindings, String
property    :application_pool, String
property    :options, String, default: ''
property    :log_directory, String, default: node['iis']['log_dir']
property    :log_period, [Symbol, String], equal_to: [:Daily, :Hourly, :MaxSize, :Monthly, :Weekly], default: :Daily, coerce: proc { |v| v.to_sym }
property    :log_truncsize, Integer, default: 1_048_576
property    :running, [true, false], desired_state: true
```

preload isn't there but it could be added

**Comment by EasyAsABC123** [comment]
>%systemroot%\system32\inetsrv\AppCmd.exe list app /xml | %windir%\system32\inetsrv\appcmd set site /in -applicationDefaults.preloadEnabled:True

> appcmd set app "websitename/vdirname" /preloadEnabled:true

> appcmd set app "Default Web Site/ApplicationName" /preloadenabled:true

**Comment by fdzedzy** [comment]
@EasyAsABC123 There is an options property on site.  The code adds any options to the string in line 119:  `cmd << " #{new_resource.options}" if new_resource.options`.

@mkeller25 Try setting that property to `"/preloadEnabled:true"`.  I think it has to be exactly what you would pass to AppCmd.exe.

**Comment by fdzedzy** [comment]
@mkeller25 You may also need to set the `/startMode:AlwaysRunning` as well.

**Comment by EasyAsABC123** [comment]
@fdzedzy lol that's what i get for skimming while gaming, thanks for calling me out 👍

**Comment by EasyAsABC123** [comment]
@fdzedzy @mkeller25 the `options` property only occurs on `add` this is because it isn't idempotent (it would update every chef client run)

**Comment by mkeller25** [comment]
Holy smokes you guys are awesome. I'm just tested "/preloadenabled:true" now and it still fails.

@fdzedzy I already use AlwaysRunning but nice callout.

I created an issue mostly because I felt it was poor documented how to use these. -limits didn't mean squat to me in the documentation, despite knowing it is also in advanced site settings in IIS.

Not sure I understand how it only runs on "add." I mean I think I do but... well I use
`action [:add, :start, :config]`

You're saying it only runs for add. So if the site is already there this won't run? Or what exactly?

Sorry!

**Comment by EasyAsABC123** [comment]
@mkeller25 correct, if the site is already there it will not run the command again.

;)

**Comment by EasyAsABC123** [comment]
try
```ruby
iis_site 'mysite.com' do
  protocol :http
  port 80
  path "#{node['iis']['docroot']}\dataverify"
  application_pool 'dataverify.com'
  options "-applicationDefaults.preloadEnabled:True"
  action [:add, :start, :config]
end
```

**Comment by mkeller25** [comment]
Here's the error from the first run, in case it helps
`[2017-06-21T16:09:26-04:00] FATAL: Mixlib::ShellOut::ShellCommandFailed: iis_site[mysite.com] (webservertest::iisconfig line 39) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0, 42], but received '1413'
cheftest01 ---- Begin output of C:\WINDOWS\System32\inetsrv\appcmd.exe add site /name:"mysite.com" /physicalPath:"C:\inetpub\wwwroot\mysite" /bindings:http/*:80: /preloadenabled:true ----
cheftest01 STDOUT: ERROR ( message:Unknown attribute "preloadenabled".  Replace with -? for help. )
cheftest01 STDERR:
cheftest01 ---- End output of C:\WINDOWS\System32\inetsrv\appcmd.exe add site /name:"mysite.com" /physicalPath:"C:\inetpub\wwwroot\mysite" /bindings:http/*:80: /preloadenabled:true ----
cheftest01 Ran C:\WINDOWS\System32\inetsrv\appcmd.exe add site /name:"mysite.com" /physicalPath:"C:\inetpub\wwwroot\mysite" /bindings:http/*:80: /preloadenabled:true returned 1413
DEBUG: [WinRM] cleaning up command_id: B3EA3FD5-1DF9-4087-9092-3C0C7328AE02 on shell_id AE7770A5-D170-4ECC-BF61-884A12315EE6
ERROR: Failed to execute command on cheftest01 return code 1`

**Comment by mkeller25** [comment]
@EasyAsABC123 that did the trick. You guys are amazing.

So...... add a little more detail to the iis_site section on options eh 👍

**Comment by EasyAsABC123** [comment]
resolved by #383
